platform-sdk/docs/hashgraph_glossary.md

```diff
@@ -0,0 +1,250 @@
+# Hashgraph Glossary
+
+<a name="top"/>
+
+- [Rounds](#rounds)
+- [Fields of an event](#fields-of-an-event)
+- [Events](#events)
+- [Event relationships](#event-relationships)
+- [Generations](#generations)
+- [Address Books](#addressbooks)
+- [Parameters](#parameters)
+
+<a name="rounds"/>
+
+## Rounds
+
+- __Pending__ round - the round number of the earliest round that hasn't yet reached consensus. This is the round whose
+  judges are currently being determined. Those consensus calculations use the roster whose roster round is the current
+  `pendingRound`. Any new event is created containing a `birthRound` field equal to the current `pendingRound`. All
+  events currently in the hashgraph should have birth rounds in the range from `pendingRound - roundsAncient` to
+  `pendingRound`, inclusive.
+- __Consensus__ round - the round number for the latest round that has reached consensus. <br>
+    - `consensusRound = pendingRound - 1`
+- __First non-ancient__ round - the oldest round that is not ancient<br>
+    - `firstNonAncientRound = pendingRound - roundsAncient`
+- __First non-expired__ round - the oldest round that is not expired<br>
+    - `firstNonExpiredRound = pendingRound - roundsExpired`
+- __Last roster__ round - Last future round for which there should be a roster in the queue<br>
+    - `lastRosterRound = pendingRound + roundsRoster`
+- __Roster__ round - Any given roster (subset of an address book) is associated with a long, which is its roster round
+  number. The `nodeID`s and consensus weights in a roster should be used when that roster round number equals the
+  pending round. A queue of rosters must be stored in state, one for each round number from `firstNonAncientRound` to
+  `lastRosterRound`, inclusive. Every time a round reaches consensus, the oldest roster can be removed. Any time the
+  transactions in a round are handled, a new roster is added for the future. An event is only allowed to be in the
+  hashgraph when its birth round is the roster round for some roster in the queue. And it must have been created by a
+  node that is listed in that roster. Of course, ancient, non-expired events may still be in memory, to gossip to nodes
+  that are behind this node, but they don't count as being in this node's hashgraph. Events with birth rounds too far in
+  the future should not be accepted during gossip. And a node should not send such events to another node, if it knows
+  that node will just discard them for that reason.
```

Comment:
I don't think we should be describing in detail how the rosters will be stored in state. This is a good conceptual explanation, but the actual implementation may be different. Can we just say that each round has a single (but not necessarily unique) roster that is used for verifying the signatures of events with a matching birth round and whose weights are used in calculating judges for that round when it is the pending round?

Reply:
There is some very good explanation of how the birth round value will influence various parts of the system in [this doc](https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md)

Reply:
+1.

Besides, it would be wasteful to keep a roster per round as rosters don't usually change that often.

I suggest to reword this definition to state that a roster that becomes active at a particular round continues to be active for all future rounds until a new roster becomes active for a future round.

This would closely correspond to the actual implementation we have to day, and probably will have in the foreseeable future.